### PR TITLE
Added entity, dao, service for lead

### PR DESCRIPTION
**Pull request by manojkrsaini09**
I have added entity and basic crud service.

### CODE REVIEW

File: src/main/java/creativei/entity/FollowUp.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private String detail;
+
+    @OneToOne
+    private Employee attendent;
+
+    @Column
+    private Integer sequence;
+
+    @OneToOne
+    private Lead lead;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/FollowUp.java]: Can not be one-to-one between follow-up and Lead. One Lead can have MANY follow-up

File: src/main/java/creativei/entity/FollowUp.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Enumerated(EnumType.STRING)
+    @Column
+    private FollowUpStatus status;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/FollowUp.java]: rename to "lastStatus"

File: src/main/java/creativei/entity/Lead.java [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    private Employee generatedBy;
+
+    @Column
+    private String description;
+
+    @Column
+    private Date nextFollowUpDate;
+
+    @OneToOne
+    private Reminder reminder;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/Lead.java]: Not one-to-one relation. One lead can have MANY reminders.

File: src/main/java/creativei/entity/LeadContact.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
+package creativei.entity;
+
+import creativei.enums.LeadTitle;
+
+import javax.persistence.*;
+
+@Entity
+@Table( name = "lead_contact")
+public class LeadContact extends BaseEntity{
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/LeadContact.java]: Lead table should be the owner of this relation. also, one Lead can have MANY Leadcontact

File: src/main/java/creativei/entity/Reminder.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package creativei.entity;
+
+import creativei.enums.ReminderType;
+
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table( name = "reminder")
+public class Reminder extends BaseEntity {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/Reminder.java]: Where is Follow-up relationship? Is this working code?

File: src/main/java/creativei/entity/LeadContact.java (hunk 4 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/entity/LeadContact.java]: Also add 
@OnetoMany
List<LeadContact> leadContacts; in Lead class

File: src/main/java/creativei/entity/Reminder.java [hunk 6]
```diff
@@ -16,6 +16,7 @@
     private String description;
 
     @ManyToOne
+    @JoinColumn(name = "lead_id")
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/Reminder.java]: Add @Column(nullable= true). Reminder can also be created without Lead.

File: src/main/java/creativei/entity/LeadContact.java (hunk 4 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/entity/LeadContact.java]: already added

**Comment by manojkrsaini09** [comment]
done

**Pull request merged**